### ISSUE TITLE
[Bug]: tailwindcss inline bg doesn't work 

### ISSUE THREAD

**Issue by mirkosaugo** [storybookjs/storybook#23271]
### Describe the bug

Hi, I just migrate storybook from v6 to v7 and i have an issue with inline background class.
`<div className={`bg-[url("/images/background.png")]`}></div>`
error:
`Module not found: Error: Can't resolve '/images/background.png' in '/Users/-/www/-/src/styles'`
(styles folder is where i load all tailwind files)
The `images` folder is inside the `public` folder. I define it in my main.js file in storybook:

`
module.exports = {
  stories: ["../src/stories/**/*.stories.mdx", "../src/stories/**/*.stories.@(js|jsx|ts|tsx)"],
  addons: [
    ...,
    {
      name: "@storybook/addon-styling",
      options: {
        // Check out https://github.com/storybookjs/addon-styling/blob/main/docs/api.md
        // For more details on this addon's options.
        postCss: {
          implementation: require("postcss"),
        },
      },
    },
    ...,
  ],
  // framework: "@storybook/react-webpack5",
  framework: {
    name: "@storybook/nextjs",
    options: {
      nextConfigPath: path.resolve(__dirname, "../next.config.js"),
    },
  },
  core: {
    builder: "webpack5",
  },
  ...,
  docs: {
    autodocs: true,
  },
};
`

In Storybook 6 everything was working fine.

**Comment by sookmax** [comment]
Hey could you try prefixing your absolute url with `/public` and see if it works? (i.e., `bg-[url("/public/images/background.png")]`).

I've encountered a similar issue and it looks like prefixing my absolute url with `/public` seems to solve the issue. Not really sure who's actually responsible for processing the `url()` specified in tailwindcss's arbitrary values (e.g., `bg-[url(...)]`), but I'm suspecting [css-loader](https://webpack.js.org/loaders/css-loader/#root) if your builder is webpack. And according to the docs, it seems `css-loader` treats absolute paths as paths relative to the _server root_. I don't know where exactly the server root the docs refers to is located, but well if we assume that's the project root, then prefixing `/public` kinda makes sense.

**Comment by mirkosaugo** [comment]
Hi @sookmax yep, I've already tried to add the prefix public and it works, but I'm using the components inside a nextJS app, so the public dir is setted. So I dont want to add the prefix, I just want to run it without issues like in the v6.

**Comment by sookmax** [comment]
@mirkosaugo You're right. It sucks and is very unintuitive having to have that `/public` prefix to make css urls work correctly. I would love to submit a PR to fix this issue myself, but I have not been able to pinpoint the exact root cause yet. It could be in `@storybook/builder-webpack5`, or could be in `@storybook/nextjs`, or who knows it might be `nextjs` itself.

I'll let you know if I learn anything useful.

Cheers.

**Comment by sookmax** [comment]
@mirkosaugo Hey, it turned out there was a conflict related to `css-loader` between `@storybook/nextjs` and `@storybook/addon-styling`. You can safely remove `options.postCss` for `@storybook/addon-styling` in `.storybook/main.ts` like below, and the `url()` should work without the `/public` prefix.

```ts
// .storybook/main.ts

...

addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@storybook/addon-onboarding',
    '@storybook/addon-interactions',
    {
      name: '@storybook/addon-styling',
      // comment out the option below, and see if url() works correctly.
      // options: {
      //   postCss: {
      //     implementation: require.resolve('postcss'),
      //   },
      // },
    },
  ],
  framework: {
    name: '@storybook/nextjs',
    options: {},
  },
  docs: {
    autodocs: 'tag',
  },

...
```

Give it a try and let me know if it works!

**Comment by mirkosaugo** [comment]
@sookmax it seems to work! Wow, so it was easier then I thought lol. Thank u so much

**Comment by pedro-stark** [comment]
@sookmax Same issue here but I'm using `"@storybook/addon-styling-webpack"`. Any idea how to solve it here?

``` javascript
// .storybook/main.js

const path = require("path");

module.exports = {
  stories: ["../src/**/*.stories.mdx", "../src/**/*.stories.@(js|jsx|ts|tsx)"],

  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-interactions",
    "@storybook/addon-styling-webpack",
    "@storybook/addon-themes",
    "@storybook/addon-storysource",
    {
      name: "@storybook/addon-styling-webpack",
      options: {
        rules: [
          {
            test: /\.css$/,
            sideEffects: true,
            use: [
              require.resolve("style-loader"),
              {
                loader: require.resolve("css-loader"),
                options: {
                  importLoaders: 1,
                },
              },
              {
                loader: require.resolve("postcss-loader"),
                options: {
                  implementation: require.resolve("postcss"),
                },
              },
            ],
          },
        ],
      },
    },
  ],

  framework: {
    name: "@storybook/react-webpack5",
    options: {},
  },

  features: {
    previewMdx2: true,
  },

  webpackFinal: async (config) => {
    config.resolve.fallback = {
      ...config.resolve.fallback,
      zlib: false,
    };

    config.resolve.modules = [
      ...(config.resolve.modules || []),
      path.resolve(__dirname, "../src"),
    ];

    // Add TypeScript path mappings
    config.resolve.alias = {
      ...config.resolve.alias,
      "@/components": path.resolve(__dirname, "../src/components"),
      "@/lib/utils": path.resolve(__dirname, "../src/lib/utils"),
      "@/lib": path.resolve(__dirname, "../src/lib"),
    };

    return config;
  },

  docs: {
    autodocs: false,
  },
};


```

**Comment by shilman** [comment]
@Integrayshaun I'm not sure what's the resolution here -- perhaps a documentation issue? -- but it would be great if you could take a look when you're back in action

**Comment by Integrayshaun** [comment]
Hi @pedro-stark 👋 

Can you share a repro of your project so that I can take a better look?
